### PR TITLE
Rebase of account get or create to accountdocumentservice

### PR DESCRIPTION
**Pull request by Ashley-Bass**

### CODE REVIEW

File: src/SFA.DAS.EAS.Portal/Application/Services/AccountDocumentService.cs [hunk 1]
```diff
@@ -14,12 +14,15 @@ public AccountDocumentService(IAccountsRepository accountsRepository)
         {
             _accountsRepository = accountsRepository;
         }
-        public Task<AccountDocument> Get(long id, CancellationToken cancellationToken = default)
+        public async Task<AccountDocument> Get(long id, CancellationToken cancellationToken = default)
```
Comment (cofaulco) [inline src/SFA.DAS.EAS.Portal/Application/Services/AccountDocumentService.cs]: Get but will create if not exists?

File: src/SFA.DAS.EAS.Portal.UnitTests/Portal/Application/Services/AccountDocumentServicesTest.cs [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+using Microsoft.ApplicationInsights.Metrics.Extensibility;
+using Microsoft.Azure.Documents.Client;
+
+namespace SFA.DAS.EAS.Portal.UnitTests.Portal.Application.Services
+{
+    [Parallelizable]
+    [TestFixture]
+    class AccountDocumentServicesTest
+    {
+        public class TestContext
```
Comment (cofaulco) [inline src/SFA.DAS.EAS.Portal.UnitTests/Portal/Application/Services/AccountDocumentServicesTest.cs]: What purpose does the test context serve here? all this logic could be done it SetUp and therefore no need for the extra class with a bunch of public properties

Comment (Ashley-Bass) [reply src/SFA.DAS.EAS.Portal.UnitTests/Portal/Application/Services/AccountDocumentServicesTest.cs]: It's the same way the other tests are structured, so I just followed the pattern, happy to switch if preferred

**Comment by cofaulco** [comment]
Old stale branch no longer required

**Pull request closed without merging**